lib/net/peer.js

```diff
@@ -264,7 +282,11 @@ class Peer extends EventEmitter {
       this.handleDrain();
     });
 
-    this.brontide.on('data', (chunk) => {
+    // Listen to the data event on brontide if
+    // encryption is enabled, otherwise just listen
+    // to the socket itself.
+    const listener = this.encrypt ? this.brontide : this.socket;
```

Comment:
You have this logic twice, I wonder if part of `init()` should be setting a single `listener` for the instance?

Reply:
Could you expand on this comment if its still relevant please?

Reply:
The variable name changed but still relevant. I was thinking its a bit more on-style for us to set an instance-level property during `init()` or `constructor()`:

`this.channel = this.encrypt ? this.brontide : this.socket;`

and then throughout the rest of the class just use `this.channel...` instead of having the switching logic in multiple places.

Reply:
I do think this is a good idea

Reply:
We would need to do some refactoring to be able to set `this.channel` in the `constructor` or `init`. Right now, I am not passing the `encrypt` property in the peer options object as to not mutate or need to copy the Pool's `this.options`

https://github.com/handshake-org/hsd/blob/733e11166cc78b6987bb89dd52f14524343ff942/lib/net/pool.js#L1112

It is set externally from the constructor:

```js
  static fromOutbound(options, addr) {
    const peer = new this(options);
    if (addr.hasKey())
      peer.encrypt = true;
```